StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PollingViewModel.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  PollingViewModel.swift
+//  StripePaymentSheet
+//
+//  Created by Fionn Barrett on 08/08/2023.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripePayments
+@_spi(STP) import StripePaymentsUI
+@_spi(STP) import StripeUICore
+import UIKit
+
+class PollingViewModel {
+
+    let paymentMethodType: STPPaymentMethodType
+    lazy var CTA: String = {
+        switch paymentMethodType {
+        case .UPI:
+            return .Localized.open_upi_app
+        default:
+            fatalError("Polling CTA has not been implemented for \(paymentMethodType)")
+        }
+    }()
+    lazy var deadline: Date = {
+        switch paymentMethodType {
+        case .UPI:
+            return Date().addingTimeInterval(60 * 5) // 5 minutes
+        default:
+            fatalError("Polling deadline has not been implemented for \(paymentMethodType)")
+        }
+    }()
+
+    init(paymentMethodType: STPPaymentMethodType) {
+        self.paymentMethodType = paymentMethodType
```

Comment:
Everything looks good, one nit that I mentioned earlier, lets fatal error if we init the view model with an unsupported payment method type. I don't think it should be possible to have an instance of the view model with a payment method type that doesn't require/support polling.